sessionctx/variable/session.go

```diff
@@ -1331,6 +1334,8 @@ type SessionVars struct {
 
 	// ETLBatchSize is the non-transactional batch size of ETL operations.
 	ETLBatchSize int
+	Stmt         ast.StmtNode
+	cacheRes     map[uint32]*CacheResult
```

Comment:
Actually, if you just cache the writePacket() data, the cache can be much easier.
It will not involve the complex ResultSet struct, and no need to consider the close / release  action of the ResultSet.